### PR TITLE
[Dynamic Region] supply extra integartion tests for `gc_worker` and `storage` in raftstore-v2.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: ref #15409 

What's Changed:

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
/cc @SpadeA-Tang @bufferflies
PTAL, thx

File: tests/integrations/server/gc_worker.rs [hunk 1]
```diff
@@ -66,3 +66,59 @@ fn test_gc_bypass_raft() {
         }
     }
 }
+
+// Raftstore-v2 introduces `Tablet` to manage regions, encapsulating the
+// operations on KVDB. This case ensures it's performed correctly.
+#[test]
+fn test_gc_bypass_raft_v2() {
+    let (cluster, leader, ctx) = test_raftstore_v2::must_new_cluster_mul(2);
```
Comment (bufferflies) [inline tests/integrations/server/gc_worker.rs:74]: why not reuse test_gc_bypass_raft?

Comment (LykxSassinator) [reply tests/integrations/server/gc_worker.rs:74]: Cannot do this directly. 
The way of getting `get_engine()` in `raftstore-v2` is different from `raftstore`

Comment (LykxSassinator) [reply tests/integrations/server/gc_worker.rs]: Refactored and removed all redundant parts after reviewing the methods supplied by `Cluster`.

File: tests/failpoints/cases/test_gc_worker.rs [hunk 2]
```diff
@@ -63,7 +64,8 @@ fn test_error_in_compaction_filter() {
 
 // Test GC worker can receive and handle orphan versions emit from write CF's
 // compaction filter correctly.
-#[test]
+#[test_case(test_raftstore::must_new_and_configure_cluster)]
+#[test_case(test_raftstore_v2::must_new_and_configure_cluster)]
 fn test_orphan_versions_from_compaction_filter() {
     let (cluster, leader, ctx) = must_new_and_configure_cluster(|cluster| {
```
Comment (SpadeA-Tang) [inline tests/failpoints/cases/test_gc_worker.rs:70]:
```suggestion
    let (cluster, leader, ctx) = new_cluster(|cluster| {
```

File: tests/failpoints/cases/test_storage.rs [hunk 3]
```diff
@@ -44,7 +45,8 @@ use tikv::{
 use tikv_util::{future::paired_future_callback, worker::dummy_scheduler, HandyRwLock};
 use txn_types::{Key, Mutation, TimeStamp};
 
-#[test]
+#[test_case(test_raftstore::new_server_cluster)]
+#[test_case(test_raftstore_v2::new_server_cluster)]
 fn test_scheduler_leader_change_twice() {
     let snapshot_fp = "scheduler_async_snapshot_finish";
     let mut cluster = new_server_cluster(0, 2);
```
Comment (SpadeA-Tang) [inline tests/failpoints/cases/test_storage.rs:52]:
```suggestion
    let mut cluster = new_cluster(0, 2);
```

**Comment by SpadeA-Tang** [comment]
`new_cluster` should be used in any case, just like
image

**Comment by LykxSassinator** [comment]
> `new_cluster` should be used in any case, just like image

got, I've misread the definition of `test_case`. I' re-check it.

**Pull request closed without merging**